package kodiak;

message Var{
  optional string name = 1;
  optional double ub = 2;
  optional double lb = 3;
}

message Param{
  optional string name = 1;
  optional double ub = 2;
  optional double lb = 3;
}

message Dfeq{
  optional string dfeq = 1;
}

message Cnstr{
 optional string cnstr = 1;
}

message Bifurcation {
  optional double res = 1;
  optional int32 prec = 2;
  repeated Var var = 3;
  repeated Param param = 4;
  repeated Dfeq dfeq = 5;
  repeated Cnstr cnstr = 6;
}

message Bif_Ans {
  repeated Box_Type boxtype = 1;
}

message Box_Type{
  repeated Box box = 1;
  optional string type_name = 2;
}

message Box{
  repeated Interval interval = 1;
}

message Interval{
  optional double lb = 1;
  optional double ub = 2;
}

message KodiakMessages{
  //Flag for speciying that matlab is done writting
  optional bool m_done = 1 [default = false];
  //Flag for specifying that kodiak is done writting
  optional bool k_done = 2 [default = false];
  repeated Bifurcation bifurcation = 3;
  repeated Bif_Ans bifans = 4;
}
  
